pytorch_to_returnn/converter/converter.py

```diff
@@ -81,17 +82,37 @@ def __init__(self,
     """
     self._model_func = model_func
     self._inputs_np = inputs
-    inputs_data_kwargs = {} if inputs_data_kwargs is None else inputs_data_kwargs.copy()
-    if "feature_dim_axis" not in inputs_data_kwargs and not inputs_data_kwargs.get("sparse", False):
-      assert len(inputs.shape) >= 2  # (batch,feature|channel,...)
-      if inputs_data_kwargs.get("batch_dim_axis", 0) == 0:
-        inputs_data_kwargs["feature_dim_axis"] = 1  # Torch uses batch-feature-major by default
-    if "shape" not in inputs_data_kwargs:
-      # Assume feature is static, all other dynamic.
-      # Assume batch-major.
-      inputs_data_kwargs["shape"] = [
-        inputs.shape[i] if i == inputs_data_kwargs["feature_dim_axis"] else None
-        for i in range(1, len(inputs.shape))]
+    assert type(self._inputs_np) in (numpy.ndarray, dict, list), (
+      "Type not supported as input: {}".format(type(inputs_data_kwargs)))
+    # inputs_data_kwargs should always be of type Dict[str|int, Dict[str, Any]].
+    # If input is array, the only key is "data". If input is a list, the keys are 0, 1, ...
+    if inputs_data_kwargs is None:
+      if isinstance(self._inputs_np, numpy.ndarray):
+        inputs_data_kwargs = {"data": {}}
+      elif isinstance(self._inputs_np, dict):
+        inputs_data_kwargs = {k: {} for k in self._inputs_np}
+      elif isinstance(self._inputs_np, list):
+        inputs_data_kwargs = {k: {} for k in range(len(self._inputs_np))}
+    if isinstance(inputs_data_kwargs, list):
+      inputs_data_kwargs = {i: x for i, x in enumerate(inputs_data_kwargs)}
+    if isinstance(self._inputs_np, numpy.ndarray) and inputs_data_kwargs.keys() != {"data"}:
+      inputs_data_kwargs = {"data": inputs_data_kwargs}
+    assert isinstance(inputs_data_kwargs, dict)
```

Comment:
This check is wrong. I assume you want to assert the same nested structure as `inputs`? Just use `nest`.